lib/jormungandr/test/bench/Latency.hs

```diff
@@ -0,0 +1,300 @@
+    extractTimings start finish . reverse <$> readTVarIO tvar
+
+-- | Scan through iohk-monitoring logs and extract time differences between
+-- start and end messages.
+extractTimings
+    :: (a -> Bool) -- ^ Predicate for start message
+    -> (a -> Bool) -- ^ Predicate for end message
+    -> [LogObject a] -- ^ Log messages
+    -> [NominalDiffTime]
+extractTimings isStart isFinish msgs = map2 mkDiff filtered
+  where
+    map2 _ [] = []
+    map2 f (a:b:xs) = (f a b:map2 f xs)
+    map2 _ _ = error "start trace without matching finish trace"
+
+    mkDiff (False, start) (True, finish) = diffUTCTime finish start
+    mkDiff (False, _) _ = error "missing finish trace"
+    mkDiff (True, _) _ = error "missing start trace"
+
+    filtered = mapMaybe filterMsg msgs
+    filterMsg logObj = case loContent logObj of
+        LogMessage msg | isStart msg -> Just (False, getTimestamp logObj)
+        LogMessage msg | isFinish msg -> Just (True, getTimestamp logObj)
+        _ -> Nothing
+    getTimestamp = tstamp . loMeta
+
+-- | Repeatedly run an action, until total elapsed time in seconds is greater
+-- than the given amount.
+repeatFor :: Int -> IO a -> IO [a]
+repeatFor nSeconds action = do
+    start <- getSystemTime
+    let repeater rs = do
+            now <- getSystemTime
+            if finished start now
+                then pure rs
+                else do
+                    !r <- action
+                    repeater (r:rs)
+    reverse <$> repeater []
+  where
+    finished a b = systemSeconds b - systemSeconds a > fromIntegral nSeconds
+
+setupLatencyLogging
+    :: TVar [LogObject ServerLog]
+    -> IO (CM.Configuration, Trace IO ServerLog)
+setupLatencyLogging tvar = do
+    cfg <- do
+        cfg' <- defaultConfigStdout
+        CM.setMinSeverity cfg' Debug
+        pure cfg'
+    pure (cfg, traceInTVarIO tvar)
+
+benchWithServer
+    :: (CM.Configuration, Trace IO ServerLog)
+    -> (Context Jormungandr -> IO ())
+    -> IO ()
+benchWithServer logCfg = withContext
+  where
+    withContext :: (Context Jormungandr -> IO ()) -> IO ()
+    withContext action = do
+        ctx <- newEmptyMVar
+        let setupContext wAddr nPort bp = do
+                let baseUrl = "http://" <> T.pack (show wAddr) <> "/"
+                let sixtySeconds = 60*1000*1000 -- 60s in microseconds
+                manager <- (baseUrl,) <$> newManager (defaultManagerSettings
+                    { managerResponseTimeout =
+                        responseTimeoutMicro sixtySeconds
+                    })
+                faucet <- initFaucet
+                putMVar ctx $ Context
+                    { _cleanup = pure ()
+                    , _manager = manager
+                    , _nodePort = nPort
+                    , _walletPort = sockAddrPort wAddr
+                    , _faucet = faucet
+                    , _feeEstimator = mkFeeEstimator (getFeePolicy bp)
+                    , _target = Proxy
+                    }
```

Comment:
Commenting after seeing another comment below about "re-using" the fee estimator:

I don't think there's any use for the fee estimator in this latency benchmark. This function is used from within a few integration scenario to test edge-cases about the fee policy. The context for the benchmark is much simpler and there's no need for this fee estimator at all.

cc @paweljakubas @rvl

Reply:
addressed in https://github.com/input-output-hk/cardano-wallet/pull/1105/commits/4c12755758ffcb713bb9ea4c9a3200de167afaee